docs/howto/rally_benchmarking.md

```diff
@@ -0,0 +1,268 @@
+│ 99th percentile service time                                   │     1050.0004142499988 ms │
+│ 100th percentile service time                                  │      1064.915250000002 ms │
+│ error rate                                                     │                    0.00 % │
+╰────────────────────────────────────────────────────────────────┴───────────────────────────╯
+
+--- Benchmark results for package: rally_benchmarks - END   ---
+Done
+```
+
+Finally, when you are done running the benchmark, bring down the Elastic Stack. 
+
+```
+elastic-package stack down
+```
+
+## Setting up an external metricstore
+
+A metricstore can be set up to send metrics collected during the benchmark execution.
+
+An external metricstore might be useful for:
+
+- Store monitoring data of the benchmark scenario for all its execution time.
+- Analyse the data generated during a benchmark. This is possible when using the `reindex-to-metricstore` flag.
+- **TODO**: Store benchmark results for various benchmark runs permanently for later comparison.
+
+In order to initialize it, you need to set up the following environment variables:
+
+```bash
+export ELASTIC_PACKAGE_ESMETRICSTORE_HOST=https://127.0.0.1:9200
+export ELASTIC_PACKAGE_ESMETRICSTORE_USERNAME=elastic
+export ELASTIC_PACKAGE_ESMETRICSTORE_PASSWORD=changeme
+export ELASTIC_PACKAGE_ESMETRICSTORE_CA_CERT="$HOME/.elastic-package/profiles/default/certs/ca-cert.pem"
+```
+
+The only one that is optional is `ELASTIC_PACKAGE_ESMETRICSTORE_CA_CERT`.
+
+When these are detected, metrics will be automatically collected every second and sent to a new index called `bench-metrics-{dataset}-{testRunID}"`.
+
+The collected metrics include the following node stats: `nodes.*.breakers`, `nodes.*.indices`, `nodes.*.jvm.mem`, `nodes.*.jvm.gc`, `nodes.*.jvm.buffer_pools`, `nodes.*.os.mem`, `nodes.*.process.cpu`, `nodes.*.thread_pool`, and `nodes.*.transport`.
+
+Ingest pipelines metrics are only collected at the end since its own collection would affect the benchmark results.
+
+You can see a sample collected metric [here](./sample_metric.json)
+
+Additionally, if the `reindex-to-metricstore` flag is used, the data generated during the benchmark will be sent to the metricstore into an index called `bench-reindex-{datastream}-{testRunID}` for further analysis. The events will be enriched with metadata related to the benchmark run.
+
+## Persisting rally tracks and dry-run
+
+If the `rally-track-output-dir` flag is used, the track and the corpus generated during the benchmark will be saved in the directory passed as value of the flag.
+Additionally, if the `dry-run` flag is used, the command will exits before running `esrally`.
+If both the flags above are used at the same time, the command will just generate the track and corpus and save them, without running any benchmark.
+If the `dry-run` flag only is used, the command will just wipe the data stream returning no report. 
```

Comment:
The rally track directory contains a few files. Can you add some short details for each file. The most important part I'm looking for: What is the rally track file that someone could copy and eventually run with esrally seperately if they want. The manual command to do it would be nice a addition.